firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TransactionTest.java

```diff
@@ -160,8 +175,20 @@ private void expectError(Code expected) {
 
     private void prepareDoc() {
       docRef = db.collection("tester-docref").document();
-      if (fromExistingDoc) {
-        waitFor(docRef.set(map("foo", "bar0")));
+
+      switch (fromDocumentType) {
+        case EXISTING:
+          waitFor(docRef.set(map("foo", "bar0")));
+          break;
+        case NON_EXISTENT:
+          // Nothing to do; document does not exist.
+          break;
+        case DELETED:
+          waitFor(docRef.set(map("foo", "bar0")));
+          waitFor(docRef.delete());
+          break;
+        default:
+          throw new Error("invalid fromDocumentType: " + fromDocumentType);
```

Comment:
`Error` in Java is rarely, if ever used directly. Use `RuntimeException` instead.

`Error` typically indicates an irrecoverable internal error inside the JVM, such as out of memory or invalid opcode.